tests/unit/amo/components/TestLandingPage.js

```diff
@@ -303,4 +278,56 @@ describe('<LandingPage />', () => {
     root.setProps({ params: { visibleAddonType: 'XUL' } });
     expect(root.find(NotFound)).toHaveLength(1);
   });
+
+  it('dispatches getLanding when category filter is set', () => {
+    const addonType = ADDON_TYPE_EXTENSION;
+
+    const { store } = dispatchClientMetadata();
+    const errorHandler = createStubErrorHandler();
+
+    // This loads a set of add-ons for a category.
+    store.dispatch(landingActions.getLanding({
+      addonType,
+      category: 'some-category',
+      errorHandlerId: errorHandler.id,
+    }));
+    store.dispatch(landingActions.loadLanding({
+      addonType,
+      featured: createAddonsApiResult([]),
+      highlyRated: createAddonsApiResult([]),
+      popular: createAddonsApiResult([]),
+    }));
+
+    const fakeDispatch = sinon.stub(store, 'dispatch');
+    render({ errorHandler, store });
+
+    sinon.assert.callCount(fakeDispatch, 2);
```

Comment:
This can go at the end of the test since the assertion message won't be so good. The other assertion messages will be more informative.

Reply:
ah right